apps/www/components/ui/toast.tsx

```diff
@@ -0,0 +1,117 @@
+import * as ToastPrimitives from "@radix-ui/react-toast"
+import { cva, type VariantProps } from "class-variance-authority"
+import { X } from "lucide-react"
+
+import { cn } from "@/lib/utils"
+
+const ToastAction: React.FC<ToastPrimitives.ToastActionProps> = ({
+  className,
+  ...props
+}) => (
+  <ToastPrimitives.Action
+    className={cn(
+      "inline-flex h-6 items-center justify-center rounded-md border border-slate-400/30 bg-transparent px-2 text-sm font-medium leading-none transition-colors hover:bg-slate-500/10 focus:outline-none focus:ring-2 focus:ring-slate-400 focus:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=open]:bg-slate-100 dark:border-slate-50/50 dark:text-slate-100 dark:hover:bg-slate-100/30 dark:hover:text-slate-100 dark:focus:ring-slate-400 dark:focus:ring-offset-slate-900 dark:data-[state=open]:bg-slate-800",
+      className
+    )}
+    {...props}
+  />
+)
+
+const ToastClose: React.FC<ToastPrimitives.ToastCloseProps> = ({
+  className,
+  ...props
+}) => (
+  <ToastPrimitives.Close
+    className={cn(
+      "absolute top-1 right-1 rounded-md p-1 hover:bg-slate-500/10 dark:hover:bg-slate-100/30",
+      className
+    )}
+    {...props}
+  >
+    <X className="h-4 w-4" />
+  </ToastPrimitives.Close>
+)
+
+const ToastTitle: React.FC<ToastPrimitives.ToastTitleProps> = ({
+  className,
+  ...props
+}) => (
+  <ToastPrimitives.Title
+    className={cn("font-semibold text-slate-900 dark:text-slate-50", className)}
+    {...props}
+  />
+)
+
+const ToastDescription: React.FC<ToastPrimitives.ToastDescriptionProps> = ({
+  className,
+  ...props
+}) => (
+  <ToastPrimitives.Description
+    className={cn("text-sm leading-none", className)}
+    {...props}
+  />
+)
+
+type ToastProps = ToastPrimitives.ToastProps
+
+const Toast: React.FC<ToastProps> = ({ className, ...props }) => (
```

Comment:
I took this PR for a spin today and got an error about forwardRef missing -- after I added the forwardRef wrapper, the error went away. 
Just wanted to share this here.

```suggestion
const Toast: React.FC<ToastProps> = React.forwardRef<HTMLLIElement, ToastProps>(
```

Reply:
I have added ref on every component that forward ref now, thanks for your feedback !